File: aoc2015/day12/src/test/java/de/havox_design/aoc2015/day12/Day12Test.java
package de.havox_design.aoc2015.day12;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day12Test {

    @SuppressWarnings("java:S2699")
    @Test
    void testMainClass() {
        MainClass.main(new String[0]);
    }

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, int expectedSum) {
        Assertions.assertEquals(expectedSum, JSONAbacus.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("day12Part1sample1.txt", 6),
                Arguments.of("day12Part1sample2.txt", 6),
                Arguments.of("day12Part1sample3.txt", 3),
                Arguments.of("day12Part1sample4.txt", 3),
                Arguments.of("day12Part1sample5.txt", 0),
                Arguments.of("day12Part1sample6.txt", 0),
                Arguments.of("day12Part1sample7.txt", 0),
                Arguments.of("day12Part1sample8.txt", 0)
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForPart2")
    void testPart2(String fileName, int expectedSum) {
        Assertions.assertEquals(expectedSum, JSONAbacus.solvePart2(fileName));
    }

    private static Stream<Arguments> getDataForPart2() {
        return Stream.of(
                Arguments.of("day12Part2sample1.txt", 6),
                Arguments.of("day12Part2sample2.txt", 4),
                Arguments.of("day12Part2sample3.txt", 0),
                Arguments.of("day12Part2sample4.txt", 6)
        );
    }
}

File: aoc2021/aoc2021-kotlin/src/test/kotlin/de/havox_design/aoc2021/day12/Day12Test.kt
package de.havox_design.aoc2021.day12

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day12Test {

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart1")
    fun testProcessPart1(filename: String, expectedResult: Int) =
        PassagePathing(filename).processPart1().shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart2")
    fun testProcessPart2(filename: String, expectedResult: Long) =
        PassagePathing(filename).processPart2().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestProcessPart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2021/day12/day12part1sample1.txt", 10),
                Arguments.of("de/havox_design/aoc2021/day12/day12part1sample2.txt", 19),
                Arguments.of("de/havox_design/aoc2021/day12/day12part1sample3.txt", 226)
            )

        @JvmStatic
        private fun getDataForTestProcessPart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2021/day12/day12part2sample.txt", 0L)
            )
    }
}
